crates/typst-pdf/src/gradient.rs

```diff
@@ -390,11 +400,21 @@ fn write_patch(
         p1, p1, p2, p2, cp1, cp2, p3, p3, p1, p1, p1, p1,
     ]));
 
-    let colors =
-        [c0.map(u16::to_be), c0.map(u16::to_be), c1.map(u16::to_be), c1.map(u16::to_be)];
-
     // Push the colors.
-    target.extend_from_slice(bytemuck::cast_slice(&colors));
+    // TODO: check if it is correct
+    target.extend_from_slice(
+        &c0.iter()
+            .flat_map(|arg0: &u16| u16::to_be_bytes(*arg0))
+            .cycle()
+            .take(4 * c0.len())
+            .chain(
+                c1.iter()
+                    .flat_map(|arg0: &u16| u16::to_be_bytes(*arg0))
+                    .cycle()
+                    .take(4 * c1.len()),
+            )
+            .collect::<SmallVec<[u8; 32]>>(),
+    );
```

Comment:
Similar approach, but simpler. I'm not sure if the `collect` is really necessary as `Vec::extend` can also just consume the iterator, but maybe it has to then resize more often as I don't think that there's a size hint.
```rs
let colors = [c0, c0, c1, c1]
    .into_iter()
    .flat_map(|c| c.iter().copied().map(u16::to_be_bytes))
    .flatten()
    .collect::<SmallVec<[u8; 32]>>();

target.extend(colors);
```

Reply:
Nice simplification again! Thanks! 
I've tentatively used the version without `collect` as it seems to indeed not be necessary. Happy to add it if it can improve performances, though.